gulp-ng-config.js

```diff
@@ -30,10 +32,24 @@ function gulpNgConfig (moduleName, configuration) {
         jsonObj,
         wrapTemplate;
 
-    try {
-      jsonObj = JSON.parse(file.contents.toString('utf8'));
-    } catch (e) {
-      this.emit('error', new PluginError(PLUGIN_NAME, 'invaild JSON file provided'));
+    if (_.endsWith(file.path, 'yml') || _.endsWith(file.path, 'yaml')) {
+      configuration.parser = 'yml';
```

Comment:
Ah yeah, just expanded the diff to see those assignments are made before the stream resource is provided.. 

We could assign the configuration defaults inside the stream handler, but that doesn't seem like a good pattern.

The other option would be to remove `parser:'json'` from the defaults and apply the logic here to determine if the user provided a parser and make the default decision at this point